--- a/src/myconfig.c
+++ b/src/myconfig.c
@@ -36,8 +36,8 @@ static const char *PACKAGE_BUGREPORT = "
 #define D_DAEMONMODE		0	/* 默认daemon模式 */
 #define D_MAXFAIL			0	/* 默认允许失败次数 */
 
-static const char *D_DHCPSCRIPT = "dhclient";	/* 默认DHCP脚本 */
-static const char *CFG_FILE = "/etc/mentohust.conf";	/* 配置文件 */
+static const char *D_DHCPSCRIPT = "udhcpc -i";	/* 默认DHCP脚本 */
+static const char *CFG_FILE = "/etc/config/mentohust.conf";	/* 配置文件 */
 static const char *LOG_FILE = "/tmp/mentohust.log";	/* 日志文件 */
 static const char *LOCK_FILE = "/var/run/mentohust.pid";	/* 锁文件 */
 #define LOCKMODE (S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH)	/* 创建掩码 */
